src/ui/public/share/directives/share_object_url.js

```diff
@@ -71,12 +74,15 @@ app.directive('shareObjectUrl', function (Private, Notifier) {
       };
 
       // since getUrl() is called within a watcher we cache the unhashing step
-      const unhashStatesInAbsUrl = memoize((absUrl) => {
-        return Private(UnhashStatesProvider).inAbsUrl(absUrl);
+      const unhashAndCacheUrl = memoize((urlWithHashes, unhashableStates) => {
```

Comment:
👍  Let's take it out and check for a performance degradation due to the watcher?